sdk/core/Azure.Core/samples/DynamicContent.md

```diff
@@ -0,0 +1,73 @@
+# Azure SDK Dynamic JSON samples
+
+Azure SDK client [Protocol methods](ProtocolMethods.md) do not take or return model types.  JSON response content can be accessed using BCL JSON types, but use of these APIs can result in code that is difficult to read and obscures the author's intent.  To improve the developer experience, Azure.Core provides a dynamic layer over JSON APIs.
```

Comment:
Note: We should add a call out regarding best practices on testing per @heaths's discovery: https://github.com/Azure/azure-sdk-for-net/pull/35320#issuecomment-1495079875

Reply:
Shouldn't we just require recorded tests already, though? Releasing an SDK without tests seems like a really bad idea. The reason I was impacted was because our Azure.Core.TestFramework doesn't check request bodies. In my case, I was switching from models to DynamicJson / DynamicData so I intentionally didn't re-record. If the records didn't match, then I did something wrong.

Using anonymous types for input was merely an implementation detail in that case.

If dev teams start with tests using dynamic, when recording them they would've found these issues. The only reason they'd miss them is if they didn't have any tests at all.

Reply:
Agreed, this addition is out of scope for these docs.